### PR TITLE
Feat(time-range-card): filter by time ranges added

### PR DESCRIPTION
**Pull request by Noam6421**
This PR implements the following feature [1208](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1208/)
In this PR I added filtering option by Time Ranges, the filter is applied both on the admin landing page and on the investigation table.

### CODE REVIEW

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                    {timeRange.displayName}
+                                </MenuItem>
+                            ))
+                        }
+                    </Select>
+                </FormControl>
+                {/* <DatePick
+                    value={new Date()}
+                    onChange={() => { }}
+                /> */}
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:66]: remove comment

File: client/src/models/enums/timeRanges.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        endDate: format(new Date(),'yyyy-MM-dd'),
+        startDate: format(subDays(new Date(), 7),'yyyy-MM-dd')
+    },
+
+    // {
+    //     id: 0,
+    //     displayName: 'טווח זמנים',
+    //     endDate,
+    //     startDate
+    // },
```
Comment (roymaniv) [inline client/src/models/enums/timeRanges.ts:36]: remove comment

File: client/src/models/FilterRulesVariables.ts [hunk 3]
```diff
@@ -1,9 +1,12 @@
+import { TimeRange } from "./TimeRange";
```
Comment (roymaniv) [inline client/src/models/FilterRulesVariables.ts:1]: single quotes

File: client/src/components/App/Content/LandingPage/adminLandingPage/useAdminLandingPage.ts [hunk 4]
```diff
@@ -4,40 +4,58 @@ import { useHistory } from 'react-router-dom';
...
-                setInvestigationInfoFilter({})
+                delete investigationInfoFilter.desks
+                setInvestigationInfoFilter((investigationInfoFilter) => {
+                    return {...investigationInfoFilter};
+                });
             }
+        } 
+        if (timeRange) {
+            setTimeRangeFilter(timeRange)
+            if(timeRange.id !== 10) {
```
Comment (roymaniv) [inline client/src/components/App/Content/LandingPage/adminLandingPage/useAdminLandingPage.ts:46]: 10 should be in const

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    });
+
+    return (
+        <LoadingCard isLoading={isLoading} className={classes.timeRangeCard}>
+            <CardContent className={classes.timeRangeCardContent}>
+                <Typography className={classes.cardTitle}>
+                    <b>{filterTimeRangeLabel}</b>
+                </Typography>
+                <FormControl variant='outlined' className={classes.timeRangeSelect}>
+                    <Select
```
Comment (y-golde) [inline client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:34]: separate into a component

**Review by y-golde — CHANGES_REQUESTED** [review]

File: client/src/models/FilterRulesVariables.ts (hunk 3 above)
Comment (Noam6421) [reply client/src/models/FilterRulesVariables.ts:1]: 👍🏼

File: client/src/models/enums/timeRanges.ts (hunk 2 above)
Comment (Noam6421) [reply client/src/models/enums/timeRanges.ts:36]: 👍🏼

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx (hunk 1 above)
Comment (Noam6421) [reply client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:66]: 👍🏼

File: client/src/components/App/Content/LandingPage/adminLandingPage/useAdminLandingPage.ts (hunk 4 above)
Comment (Noam6421) [reply client/src/components/App/Content/LandingPage/adminLandingPage/useAdminLandingPage.ts:46]: 👍🏼

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                            },
+                            transformOrigin: {
+                                vertical: "top",
+                                horizontal: "left"
+                            },
+                            getContentAnchorEl: null
+                        }}
+                        value={timeRangeFilter.id}
+                        onChange={(event) => {
+                            onTimeRangeChange(event.target.value as number)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:48]: you dont need the curly braces

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import UpdateButton from '../UpdateButton/UpdateButton';
+import useTimeRangeFilterCard from './useTimeRangeFilterCard';
+import AdminLandingPageFilters from '../AdminLandingPageFilters';
+
+const filterTimeRangeLabel = 'טווח זמנים';
+
+const TimeRangeCard = (props : Props): JSX.Element => {
+
+    const classes = useStyles();
+    const { timeRangeFilter, setTimeRangeFilter, investigationInfoFilter, setInvestigationInfoFilter } = props;
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:19]: enter

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    const { isLoading, onTimeRangeChange, onUpdateButtonCLicked } = useTimeRangeFilterCard({
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx:20]: enter

File: client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx (hunk 5 above)
Comment (Noam6421) [reply client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/timeRangeFilterCard.tsx]: 👍🏼

**Review by roymaniv — APPROVED** [review]

**Review by y-golde — APPROVED** [review]

**Pull request merged**